src/runtime/Memory.h

```diff
@@ -74,14 +74,19 @@ class Memory : public Extern {
     template <typename T>
     void load(ExecutionState& state, uint32_t offset, uint32_t addend, T* out) const
     {
-        checkAccess(state, offset, addend, sizeof(T));
+        if (!checkAccess(offset, addend, sizeof(T))) {
+            throwException(state, offset, addend, sizeof(T));
```

Comment:
Why some methods of `Memory` still throw an exception?
The other functions such as `Memory::fill` is modifed to return a boolean value instead of throwing an exception.

Reply:
These are not used by jit. Maybe there could be a `checkAccess` variant which works in the old way (no inline).

Reply:
@zherczeg 
This confuses me because some methods throw exception while others return a boolean 😕 
IMHO its better to unify the exception handling in interpreter and JIT both
What do you think about it?

Reply:
I would not change the current working of the interpreter, because c++ exception handling is better than returning true/false. The problem is, each compiler has its own exception handling implementation, and supporting all compilers and os-es in the word is too difficult, so jit will do this differently, but only when jit code is running.